src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CallbackHandlerFactory.java

```diff
@@ -11,20 +12,30 @@
 import java.util.List;
 import java.util.Optional;
 
+import static uk.gov.hmcts.cmc.claimstore.services.ccd.callbacks.CallbackParams.Params.BEARER_TOKEN;
+
 @Service
 public class CallbackHandlerFactory {
 
+    private final UserRolesService userRolesService;
     private HashMap<String, CallbackHandler> eventHandlers = new HashMap<>();
 
     @Autowired(required = false)
-    public CallbackHandlerFactory(List<CallbackHandler> beans) {
+    public CallbackHandlerFactory(List<CallbackHandler> beans, UserRolesService userRolesService) {
+        this.userRolesService = userRolesService;
         beans.forEach(bean -> bean.register(eventHandlers));
     }
 
     public CallbackResponse dispatch(CallbackParams callbackParams) {
         return Optional.ofNullable(eventHandlers.get(callbackParams.getRequest().getEventId()))
+            .filter(h -> hasSupportedRoles(h, callbackParams.getParams().get(BEARER_TOKEN).toString()))
```

Comment:
if the passed token doesn't have the authorised role, we should throw the relevant exception/Http status back.

Reply:
done as suggested

Reply:
Thanks 👍